workspaces/tech-insights/plugins/tech-insights-maturity-common/package.json

```diff
@@ -0,0 +1,64 @@
+{
+  "name": "@backstage-community/plugin-tech-insights-maturity-common",
+  "description": "Common functionalities for the tech-insights-maturity plugin",
+  "version": "0.1.0",
+  "main": "./dist/index.cjs.js",
+  "types": "./dist/index.d.ts",
+  "license": "Apache-2.0",
+  "private": true,
+  "publishConfig": {
+    "access": "public"
+  },
+  "exports": {
+    ".": "./src/index.ts",
+    "./package.json": "./package.json"
+  },
+  "typesVersions": {
+    "*": {
+      "package.json": [
+        "package.json"
+      ]
+    }
+  },
+  "backstage": {
+    "role": "common-library",
+    "pluginId": "tech-insights",
+    "pluginPackages": [
+      "@backstage-community/plugin-tech-insights-maturity",
+      "@backstage-community/plugin-tech-insights-maturity-common"
+    ]
+  },
+  "keywords": [
+    "backstage",
+    "tech-insights",
+    "tech-insights-maturity"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/backstage/community-plugins",
+    "directory": "workspaces/tech-insights/plugins/tech-insights-maturity-common"
+  },
+  "sideEffects": false,
+  "scripts": {
+    "build": "backstage-cli package build",
+    "lint": "backstage-cli package lint",
+    "test": "backstage-cli package test",
+    "clean": "backstage-cli package clean",
+    "prepack": "backstage-cli package prepack",
+    "postpack": "backstage-cli package postpack"
+  },
+  "dependencies": {
+    "@backstage-community/plugin-tech-insights-backend-module-jsonfc": "workspace:^",
```

Comment:
Would it be helpful as a follow up to move some of the shared utils and types to a new `jsonfc-common` package? I don't think it's a good practice to import a modules into common library. I could be wrong 🫠

Reply:
Hmm you may be right. @Xantier what do you think?

 Maybe it is even okay for me to add it directly to the jsonfc package...

These are the lines where I needed these dependencies. 
```
export interface MaturityCheck extends TechInsightCheck {
  metadata: {
    category: string;
    rank: 0 | 1 | 2 | 3;
    solution: string;
  };
}
```

```
/**
 * MaturityJsonRuleCheck implementation of TechInsightJsonRuleCheck
 * @public
 */
export interface MaturityJsonRuleCheck extends TechInsightJsonRuleCheck {
  metadata: {
    category: string;
    rank: 0 | 1 | 2 | 3; // 0 - Stone, 1 - Bronze, 2 - Silver, 3 - Gold
    solution: string;
  };
}
```

Reply:
Decided to just remove these dependencies inside tech-insights-maturity-common as I am not using it anywhere in this plugin. It's more for type-checking when creating declaring the checks so that they don't forget to add the Maturity metadata but should be fine without it...

Reply:
It's probably worth having these items typed in here. I believe the upper item can be modified to use the `-common` tech insights package now with @kurtaking's changes. The lower item wouldn't ideally import the `-json-fc` since that contains logic that we don't want to depend on on this common package. 

I'd maybe consider that we should make the metadata generically typable on the higher level check type and allow this (and possibly other) extension packages to provide just that type. That can then be typed on the usage site, achieving the original goal of allowing more strictness. 

For this, if you could add the upper item back in, let's leave the lower out for now. It is a lesser evil to omit it than adding a dependency which usually ends up being more difficult to remove.